collector/src/p2p/engine.rs

```diff
@@ -105,16 +107,16 @@ where
     /// Returns a handle that can be used to wait for the engine to complete.
     pub fn start(
         mut self,
-        requests: (impl Sender<PublicKey = P>, impl Receiver<PublicKey = P>),
-        responses: (impl Sender<PublicKey = P>, impl Receiver<PublicKey = P>),
+        requests: (S, impl Receiver<PublicKey = P>),
+        responses: (S, impl Receiver<PublicKey = P>),
     ) -> Handle<()> {
         self.context.spawn_ref()(self.run(requests, responses))
     }
 
     async fn run(
         mut self,
-        requests: (impl Sender<PublicKey = P>, impl Receiver<PublicKey = P>),
-        responses: (impl Sender<PublicKey = P>, impl Receiver<PublicKey = P>),
+        requests: (S, impl Receiver<PublicKey = P>),
```

Comment:
It was purposeful to keep this as `impl Sender` here 👍 

It makes it much harder to instantiate in practice if we have to bind the type to the struct during init.

Reply:
If we rethought this entrypoint, I think we could then actually bind these types to the struct.